### DISCUSSION TITLE
Build Failed because of package loading issue

### DISCUSSION THREAD

**Discussion by a1exkita** [jtr13/cc21fall1#53] [Q&A] [answered]
Hello @jtr13 @s10singh97 ,

There is a build failed error because of package issue on the merged pull request #41 .
What is the exact procedure when we face errors when it is deploying?

Also, I just realized that I am not supposed to merge it by myself. Sorry about that.

Here's the error message (as you can see on GitHub Actions):
```
Quitting from lines 85-94 (caret_machine_learning_tutorial.Rmd) 
Error: Error: package RANN is required
Execution halted
Error in Rscript_render(f, render_args, render_meta, add1, add2) : 
  Failed to compile caret_machine_learning_tutorial.Rmd
Calls: <Anonymous> -> render_new_session -> Rscript_render
Execution halted
Error: Process completed with exit code 1.
```

Thanks in advance!

**Comment by a1exkita** [comment]
I tried to knit the Rmd file on my local machine, but it did not work out because of the same package error. Also, based on the building error message, I added `library(RANN)` to the script on my local, but it generated some different errors. Therefore, it seems like the Rmd file had some issues when the authors created the pull request, which we can't detect in the current merging procedure.

One way we can resolve this issue is the following:
- ask the authors to fix this building issue on their end
- In the meantime, remove their file and filename from _bookdown.yml on main branch so that we can proceed with other merging tasks 
- once the authors fix the building issue, they resubmit the pull request

Please tell me if there are some issues on this process @jtr13 @s10singh97 .
I will do the above once it's approved.

Thank you,
Alex

**Reply by s10singh97** [reply]
Yes, the build is failing because of the internal errors of the code. We can just comment this file's name from the `_bookdown.yml` file.

**Reply by a1exkita** [reply]
Will do! Should we let the authors know that there is some compilation error in their code and ask to resubmit another pull request? @jtr13

**Comment by jtr13** [comment, marked as answer]
Or you can add the RANN package to the DESCRIPTION file.  And if it still doesn't work (likely scenario) add `eval=FALSE` to `knitr::opts_chunk$set` at the top of that chapter and ask the authors to fix their code.

**Reply by a1exkita** [reply]
I added the RANN package to the DESCRIPTION, and it did not compile successfully. So, I have added eval=FALSE and asked them to fix the issue. Thank you so much for the help!

**Reply by a1exkita** [reply]
Just FYI, since they modified the code and resubmitted another PR, I successfully merged it to the main branch 🚀🔥 Thank you, Shashwat @s10singh97 and Professor Robbins @jtr13 !
